src/config.ts

```diff
@@ -10,68 +10,26 @@
  * All portions of the code written by Atypon Systems LLC are Copyright (c) 2019 Atypon Systems LLC. All Rights Reserved.
  */
 
-const isTrue = (value: string | undefined) => {
-  return value === '1' || value === 'true'
-}
-
-const normalizeURL = (url: string | undefined) => {
-  return url && url.replace(/\/$/, '')
-}
-
-interface Config {
+export interface Config {
   environment: string
-  analytics: {
-    id?: string
-  }
   api: {
     url: string
     headers: Record<string, unknown>
   }
   features: {
     footnotes: boolean
     fileManagement: boolean
-    commenting: boolean
-    DOI: boolean
-    runningTitle: boolean
     tableEditing: boolean
     pullQuotes: boolean
   }
-  support: {
-    email: string
-  }
-  quarterback: {
-    enabled: boolean
-    url: string
-  }
-  keywordsCategories: boolean
 }
 
-const config = {
-  environment: process.env.NODE_ENV,
-  analytics: {
-    id: process.env.GOOGLE_ANALYTICS_ID,
-  },
-  api: {
-    url: normalizeURL(process.env.MANUSCRIPTS_API_URL),
-    headers: {},
-  },
-  features: {
-    footnotes: isTrue(process.env.FEATURE_FOOTNOTES),
-    fileManagement: isTrue(process.env.FEATURE_FILE_MANAGEMENT),
-    commenting: isTrue(process.env.FEATURE_COMMENTS),
-    DOI: isTrue(process.env.FEATURE_DOI),
-    runningTitle: isTrue(process.env.FEATURE_RUNNING_TITLE),
-    tableEditing: isTrue(process.env.FEATURE_TABLE_EDITING),
-    pullQuotes: isTrue(process.env.FEATURE_PULL_QUOTES),
-  },
-  support: {
-    email: process.env.SUPPORT_EMAIL || 'support@manuscriptsapp.com',
-  },
-  quarterback: {
-    enabled: isTrue(process.env.QUARTERBACK_ENABLED),
-    url: normalizeURL(process.env.QUARTERBACK_URL) || '',
-  },
-  keywordsCategories: isTrue(process.env.FEATURE_KEYWORDS_CATEGORIES),
+let config: Config
```

Comment:
Can we store the config in react state or context?

Reply:
2 reasons why I didn't use Context:

- I don’t expect the configuration to change, so using Context would introduce unnecessary complexity.
- Context cannot be directly used in class constructors or non-React code.

Reply:
Thank you, Mikhail, for bringing these points up. I agree that ensuring data immutability would be beneficial, and I hadn't considered that aspect. Your suggestion about relying on the React ecosystem to manage that and ensure immutability makes sense, but the idea that context cannot be directly used in non-React code will be very annoying, especially for the configuration case, and we will need to drill props along in many places (e.g., see `methods.ts`). Do you have any thoughts or suggestions on how to tackle that if we stick with using context?

Reply:
Seems like I'll stick with the current implementation, but I'll update it to prevent changes to the config variable once it's set up for the first time (as you mentioned in your first point). Any objections?

Reply:
Sounds good, thanks Islam.